http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java

```diff
@@ -473,6 +485,36 @@ public HttpServerConfig setCompressionEnabled(boolean compressionEnabled)
         return this;
     }
 
+    @MinDataSize("128MB")
+    @MaxDataSize("32GB")
+    public DataSize getMaxHeapMemory()
+    {
+        return maxHeapMemory;
+    }
+
+    @Config("http-server.max-heap-memory")
+    @ConfigHidden
+    public HttpServerConfig setMaxHeapMemory(DataSize maxHeapMemory)
+    {
+        this.maxHeapMemory = maxHeapMemory;
+        return this;
+    }
+
+    @MinDataSize("128MB")
+    @MaxDataSize("32GB")
```

Comment:
also this

Reply:
the default would reach this value if the heap size is 245 GB

Reply:
Droprd